insights/parsers/zdump_v.py

```diff
@@ -0,0 +1,122 @@
+"""
+ZdumpV - command ``/usr/sbin/zdump -v /etc/localtime -c 2019,2039``
+===================================================================
+
+The ``/usr/sbin/zdump -v /etc/localtime -c 2019,2039`` command provides information about
+'Daylight Saving Time' in file /etc/localtime from 2019 to 2039.
+
+Sample content from command ``zdump -v /etc/localtime -c 2019,2039`` is::
+    /etc/localtime  Sun Mar 10 06:59:59 2019 UTC = Sun Mar 10 01:59:59 2019 EST isdst=0 gmtoff=-18000
+    /etc/localtime  Sun Mar 10 07:00:00 2019 UTC = Sun Mar 10 03:00:00 2019 EDT isdst=1 gmtoff=-14400
+    /etc/localtime  Sun Nov  7 05:59:59 2038 UTC = Sun Nov  7 01:59:59 2038 EDT isdst=1 gmtoff=-14400
+    /etc/localtime  Sun Nov  7 06:00:00 2038 UTC = Sun Nov  7 01:00:00 2038 EST isdst=0 gmtoff=-18000
+
+Examples:
+    >>> dst = zdump.data[0]
+    >>> dst.get('utc_time')
+    datetime.datetime(2019, 3, 10, 6, 59, 59)
+    >>> dst.get('utc_time_raw')
+    'Sun Mar 10 06:59:59 2019 UTC'
+    >>> dst.get('local_time')
+    datetime.datetime(2019, 3, 10, 1, 59, 59)
+    >>> dst.get('local_time_raw')
+    'Sun Mar 10 01:59:59 2019 EST'
+    >>> dst.get('isdst')
+    False
+    >>> dst.get('gmtoff')
+    -18000
+"""
+
+from datetime import datetime
+from insights.specs import Specs
+from insights.parsers import SkipException
+from insights import parser, CommandParser
+
+
+def str2datetime(timestamp, tz=False):
+    """
+    This function translates the time stamp into a datetime object.
+
+    Args:
+        timestamp (str): the time stamp from command `zdump -v`
+        tz (bool): True if it's UTC TimeZone.
+
+    Returns:
+        time (datetime): the datetime object about the time stamp
+        time_string (str): the formatted time stamp met the `TimeStamp Specification`
+    """
+
+    # The gap between fields will be changed in command `zdump`
+    # So we split it into slice and format output in the specification
+    # fs stands for 'fields'
+    fs = [s for s in timestamp.split(' ') if len(s) > 0]
+    if len(fs) != 6:
+        return
+
+    # `Weekday` `Month` `Day of the month` `Hour:Minute:Second` `Year` `TimeZone`
+    time_string = "%s %s %2s %s %s %s" % (fs[0], fs[1], fs[2], fs[3], fs[4], fs[5])
+
+    # Fixed the problem that the program running this python code doesn't
+    # has the corresponding TimeZone where strptime will raise ValueError.
+    # So, we skip the `TimeZone`
+    time_s = time_string.rsplit(None, 1)[0]
+    time_f = "%a %b %d %H:%M:%S %Y"
+
+    if tz:
+        # In some version, `zdump` prints 'UT' instead of 'UTC'
+        # 'UC' is an invalid TimeZone for function `strptime`
+        time_s = time_s + " UTC"
+        time_f = "%a %b %d %H:%M:%S %Y %Z"
+
+    try:
+        time = datetime.strptime(time_s, time_f)
+    except ValueError:
+        pass
+
+    return time, time_string
```

Comment:
Here the `time_string` is not the `RAW` string since you have re-formatted it in L57